pymeasure/instruments/hp/hp8753e.py

```diff
@@ -0,0 +1,394 @@
+        else:
+            raise Exception("Invalid scattering parameter requested" " for Hewlett Packard 8753E")
+
+    @property
+    def scan_points(self):
+        """Gets the number of scan points"""
+        search = re.search(r"\d\.\d+E[+-]\d{2}$", self.ask("POIN?"), re.MULTILINE)
+        if search:
+            return int(float(search.group()))
+        else:
+            raise Exception("Improper message returned for the" " number of points")
+
+    @scan_points.setter
+    def scan_points(self, points):
+        """Sets the number of scan points, truncating to an allowed
+        value if not properly provided
+        """
+        points = discreteTruncate(points, HP8753E.SCAN_POINT_VALUES)
+        if points:
+            self.write("POIN%d" % points)
+        else:
+            raise RangeException("Maximum scan points (1601) for" " Hewlett Packard 8753E exceeded")
+
+    @property
+    def IFBW(self):
+        """Gets the IF Bandwidth"""
+        search = re.search(r"\d\.\d+E[+-]\d{2}$", self.ask("IFBW?"), re.MULTILINE)
+        if search:
+            return int(float(search.group()))
+        else:
+            raise Exception("Improper message returned for the" " IF Bandwidth")
+
+    @IFBW.setter
+    def IFBW(self, bandwidth):
+        """Sets the resolution bandwidth (IF bandwidth)"""
+        allowedBandwidth = [10, 30, 100, 300, 1000, 3000, 3700, 6000]
+        bandwidth = discreteTruncate(bandwidth, allowedBandwidth)
+        if bandwidth:
+            self.write("IFBW%d" % bandwidth)
+        else:
+            raise RangeException(
+                "Maximum IF bandwidth (6000) for Hewlett Packard " "8753E exceeded"
+            )
+
+    def set_averaging(self, averages):
+        """Sets the number of averages and enables/disables averaging. Should be
+        between 1 and 999"""
+        averages = int(averages)
+        if not 1 <= averages <= 999:
+            assert RangeException("Set", averages, "must be in the range 1 to 999")
+        self.averages = averages
+        self.averaging_enabled = averages > 1
+
+    def disable_averaging(self):
+        """Disables averaging"""
+        warnings.warn(
+            "Don't use disable_averaging(), use averaging_enabled = False instead",
+            FutureWarning,
+        )
+        self.averaging_enabled = False
+
+    def enable_averaging(self):
+        """Enables averaging"""
+        warnings.warn(
+            "Don't use enable_averaging(), use averaging_enabled = True instead",
+            FutureWarning,
+        )
+        self.averaging_enabled = True
+
+    def is_averaging(self):
+        """Returns True if averaging is enabled"""
+        warnings.warn("Don't use is_averaging(), use averaging_enabled instead", FutureWarning)
+        return self.averaging_enabled
+
+    def restart_averaging(self, averages):
+        warnings.warn("Don't use restart_averaging(), use scan_single() instead", FutureWarning)
+        self.scan_single()
```

Comment:
I figure you copied these methods from the other driver.

As they are deprecated, remove them, please.